Jetpack compose

PavlosFas13 opened the pull request:
New pull request, to start the home page

Commit:
v5

Commit:
small update

Commit:
v6

Commit:
small update

Commit:
javadoc utf-8 encoding fix

diamantidakos committed:
Merge pull request #4 from NicholasSarantinos/android

Android Pull Request #1

Commit:
v7

Commit:
V F1

diamantidakos committed:
Merge pull request #5 from NicholasSarantinos/android

Android

Commit:
Release

Commit:
small update

diamantidakos committed:
Merge pull request #6 from NicholasSarantinos/android

Android

bzafiris committed:
Update README.md

bzafiris committed:
Migrated to android 12

bzafiris committed:
Updated documentation and project name

bzafiris committed:
Removed libs directory

bzafiris committed:
Added book search activity with ViewModel support

bzafiris committed:
Added reservation activity

bzafiris committed:
Added book reservation activity basic implementation

bzafiris committed:
Added reservation request domain logic

bzafiris committed:
Added activity interaction

bzafiris committed:
Added tests for domain logic

bzafiris committed:
Added unit tests for BookReservationPresenter

bzafiris committed:
Added unit tests for BookSearchPresenter

bzafiris committed:
Code cleanup for book search activity

bzafiris committed:
Added sequence diagram for Book Reservation request

bzafiris committed:
Migrated to latest AGP and ui libraries

bzafiris committed:
Added jetpack compose dependencies

bzafiris committed:
Added jetpack compose dependencies

bzafiris committed:
Migrated theme (partially) to compose

bzafiris committed:
Presented compose content on Java activity

bzafiris committed:
Resolved conflicts

PavlosFas13 committed:
HomePage

I created the HomePageActivity.kt file, and changed AndroidManifest in order to be this file the HomePage of the app.

bzafiris commented on app/src/main/java/com/mgiandia/library/ui/composable/HomePageActivity.kt:
```diff
@@ -0,0 +1,151 @@
+
+        Row()
+        {
+            displayButton("Δάνεια")
```
@PavlosFas13 είναι καλή πρακτική να αντικαταστήσεις τα strings με string resources (βάζεις το string στο strings.xml και αναφέρεις εδώ το id του).

bzafiris committed:
Call composable from HomePageActivity

PavlosFas13 committed:
Home page

I made the home page, but I don't know how to find the button from java, and set onClickListener to send it to kotlin class.

bzafiris committed:
Upgraded AGP plugin

bzafiris committed:
Added button handling with LiveData

PavlosFas13 committed:
Add edit book and add edit authors pages

PavlosFas13 committed:
Small updates

PavlosFas13 committed:
Merge branch 'jetpack-compose' of https://github.com/PavlosFas13/Library into jetpack-compose
